Docker/qgis3.4.5/Dockerfile

```diff
@@ -0,0 +1,93 @@
+FROM ubuntu:xenial
+MAINTAINER Richard Boon <richard.boon@nelen-schuurmans.nl>
+# Inspired by https://hub.docker.com/r/timcera/qgis-desktop-ubuntu/~/dockerfile/
+
+ENV DEBIAN_FRONTEND noninteractive
+ENV DEBCONF_NONINTERACTIVE_SEEN true
+
+ENV TZ Europe/Amsterdam
+
+ARG USER_ID
+ARG GROUP_ID
+ARG USERNAME
+
+RUN echo $TZ > /etc/timezone \
+    && apt-get -y update \
+    && apt-get -y install --no-install-recommends \
+        tzdata \
+        dirmngr \
+        apt-transport-https \
+        python-software-properties \
+        software-properties-common \
+        wget \
+        dbus \
+        git \
+        gosu \
+    && add-apt-repository ppa:ubuntugis/ubuntugis-unstable \
+    && rm /etc/localtime \
+    && ln -snf /usr/share/zoneinfo/$TZ /etc/localtime \
+    && dpkg-reconfigure -f noninteractive tzdata \
+    && apt-get clean \
+    && apt-get purge \
+    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+RUN dbus-uuidgen > /var/lib/dbus/machine-id
+
+RUN echo "deb     https://qgis.org/ubuntugis-ltr xenial main" >> /etc/apt/sources.list
+RUN echo "deb-src https://qgis.org/ubuntugis-ltr xenial main" >> /etc/apt/sources.list
+RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-key CAEB3DC3BDF7FB45
+
+RUN apt-get -y update \
+    && apt-get -y install --no-install-recommends \
+        python3-dev \
+        python3-pip \
+        gdal-bin \
+        qgis \
+        python-qgis \
+        qgis-provider-grass \
+        grass \
+        pyqt5-dev-tools \
+        libhdf5-serial-dev \
+        python-dev \
+        build-essential \
+    && apt-get clean \
+    && apt-get purge \
+    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+RUN pip3 install --upgrade pip
+RUN pip3 install setuptools
+
+ADD requirements.txt .
+ADD requirements-dev.txt .
+RUN pip3 install -r requirements.txt -U
+RUN pip3 install -r requirements-dev.txt -U
+
+# Remove once netcdf4 is completely removed.
+ENV USE_SETUPCFG=0
+ENV HDF5_INCDIR=/usr/include/hdf5/serial
+ENV HDF5_LIBDIR=/usr/lib/x86_64-linux-gnu/hdf5/serial
+RUN pip install netCDF4
+
+# Make sure the h5py is compiled with the same HDF5 version of QGIS.
+RUN pip3 uninstall h5py -y
+ENV HDF5_DIR /usr/lib/x86_64-linux-gnu/hdf5/serial/
+RUN pip3 install --no-binary=h5py h5py
+
+RUN groupadd -g $GROUP_ID $GROUP_ID
+RUN useradd -m -d /home/$USERNAME --shell /bin/bash --uid $USER_ID --gid $GROUP_ID $USERNAME
```

Comment:
add '-l' for large user id / group id ? :))

RUN useradd -m -d -l /home/$USERNAME --shell /bin/bash --uid $USER_ID --gid $GROUP_ID $USERNAME